terracotta/drivers/raster_base.py

```diff
@@ -36,6 +38,42 @@
 logger = logging.getLogger(__name__)
 
 
+class LFUCacheWithCompression(LFUCache):
+    """ Least-frequently-used Cache with data compression
+    """
+
+    def __getitem__(self, key: Any) -> np.ma.MaskedArray:
+        compressed_item = super().__getitem__(key)
+        return self._decompress_masked_array(compressed_item)
+
+    def __setitem__(self, key: Any, value: np.ma.MaskedArray) -> None:
+        super().__setitem__(key, self._compress_masked_array(value))
+
+    def _compress_masked_array(self, array: np.ma.MaskedArray) -> Tuple:
+        compressed_data = zlib.compress(array.data)
+        mask_to_int = np.packbits(array.mask.astype(np.uint8))
+        compressed_mask = zlib.compress(mask_to_int)
+        return (compressed_data,
+                compressed_mask,
+                array.dtype.name,
+                array.shape
+                )
+
+    def _decompress_masked_array(self, data: Tuple) -> np.ma.MaskedArray:
+        dt = np.dtype(data[2])
+        shape = data[3]
+        decompressed_data = np.frombuffer(zlib.decompress(data[0]), dtype=dt).reshape(shape)
+        decompressed_mask_packed = np.frombuffer(zlib.decompress(data[1]), dtype=np.uint8)
+        decompressed_mask_unpacked = np.unpackbits(decompressed_mask_packed)[:np.prod(shape)]
+        decompressed_mask = decompressed_mask_unpacked.reshape(shape)
```

Comment:
Not super important, but I think it would be cleaner to re-use the same variable for decompression of the mask:
```python
decompressed_mask = np.frombuffer(zlib.decompress(data[1]), dtype=np.uint8)
decompressed_mask = np.unpackbits(decompressed_mask)[:np.prod(shape)]
decompressed_mask = decompressed_mask.reshape(shape)
```
This also has the benefit of re-using memory.

Reply:
maybe even just call it `mask`